README.md

```diff
@@ -18,4 +20,167 @@ Execute the following steps to practice various topics covered in the intro to g
+| Creates a new commit that combines the changes from two branches | Reapplies the commits from one branch on top of another branch |
+| Creates a non-linear commit history                              | Creates a linear commit history                                |
+| Preserves the commit history of both branches                    | Rewrites the commit history of the branch being rebased |
+
+Now that you have rebased the `mySecondBranch` branch on top of the `main` branch, use `git log` to compare the commit history with the one where you merged `myFirstBranch` with `main`.
+
+## Cherry-pick
+
+`git cherry-pick` is used to apply the changes introduced by some existing commits. It is useful when you want to apply a single commit from one branch to another.
+
+At this point, if you are on `main`, you will **not** have the `new2.txt` and `new3.txt` files.
+
+If you switch to `mySecondBranch`, you will have the `new2.txt` and `new3.txt` files as well as the `new4.txt` file.
+
+Now, consider the following scenario:
+
+You want the `new3.txt` file to be added to the `main` branch, but you do not want the `new2.txt` file to be added.
+
+To achieve this, you can use the `git cherry-pick` command.
+
+1) Use `git log` to find the commit hash of the commit where you added `new3.txt`.
+
+    In the example below, the commit hash is `c8789dd`.
+
+    ![Commit Log 2](/content/images/log_rebase_2.png)
+
+    Note: The commit hash will be different in your case.
+
+2) Make sure you are on the `main` branch and use the following command to cherry-pick the commit:
+
+    ```bash
+    git cherry-pick <<commit_hash>>
+    ```
+
+3) Verify that the changes have been applied to the `main` branch by using the `git log` command. You can also verify this by checking that you now have the `new3.txt` file in the `main` branch, but not the `new2.txt` file.
+
+    ![Commit Log 3](/content/images/log_cherrypick_1.png)
+
+## Interactive Rebase
+
+`git rebase -i` is used to interactively rebase commits. This is useful when you want to modify, delete, or combine commits.
+
+Consider the following scenario:
+
+1) Switch to `mySecondBranch`. Add a new text file called `new5.txt` and add some text to it. Add and commit these changes.
+2) Edit the `new2.txt` file and add some more text to it. Add and commit these changes.
+3) Now, you should have the following commit history:
+
+    ![Commit Log 4](/content/images/log_irebase.png)
+
+4) You want to rebase `mySecondBranch` onto `main`, but you have already cherry-picked the commit that added `new3.txt` to the `main` branch.
+
+    Now, you want the `new2.txt` file in your main branch, but it has two commits associated with it. You want to combine these two commits into a single commit. Moreover, now you do not want the `new5.txt` file anymore.
+
+    Checkout to `mySecondBranch` and use the following command to interactively rebase the commits:
+
+    ```bash
+    git rebase -i main
```

Comment:
Include this statement as well ```git rebase -i HEAD~3``` before the rebasing onto main example. You don't have to write all the steps as they are all the same. Just mention a line or two about what this does, and when it is used

Reply:
ok, will do